src/processes.c

```diff
@@ -1574,42 +1572,57 @@ static char *ps_get_cmdline(long pid, char *name, char *buf, size_t buf_len) {
   return buf;
 } /* char *ps_get_cmdline (...) */
 
-static int read_fork_rate(void) {
-  FILE *proc_stat;
-  char buffer[1024];
+static int read_fork_rate(const char *buffer) {
   value_t value;
-  bool value_valid = 0;
+  char id[] = "processes ";
+  char *processes;
 
-  proc_stat = fopen("/proc/stat", "r");
-  if (proc_stat == NULL) {
-    ERROR("processes plugin: fopen (/proc/stat) failed: %s", STRERRNO);
+  int status;
+  char *fields[2];
+  int fields_num;
+
+  processes = strstr(buffer, id);
+  if (!processes) {
+    WARNING("'processes ' not found in /proc/stat");
     return -1;
   }
 
-  while (fgets(buffer, sizeof(buffer), proc_stat) != NULL) {
-    int status;
-    char *fields[3];
-    int fields_num;
+  fields_num = strsplit(processes, fields, STATIC_ARRAY_SIZE(fields));
+  if (fields_num != 2)
+    return -1;
 
-    fields_num = strsplit(buffer, fields, STATIC_ARRAY_SIZE(fields));
-    if (fields_num != 2)
-      continue;
+  status = parse_value(fields[1], &value, DS_TYPE_DERIVE);
+  if (status != 0)
+    return -1;
 
-    if (strcmp("processes", fields[0]) != 0)
-      continue;
+  ps_submit_global_stat("fork_rate", value.derive);
+  return 0;
+}
 
-    status = parse_value(fields[1], &value, DS_TYPE_DERIVE);
-    if (status == 0)
-      value_valid = 1;
+static int read_sys_ctxt_switch(const char *buffer) {
+  value_t value;
+  char id[] = "ctxt ";
+  char *ctxt;
+
+  int status;
+  char *fields[2];
+  int fields_num;
 
-    break;
+  ctxt = strstr(buffer, id);
+  if (!ctxt) {
+    WARNING("'ctxt ' not found in /proc/stat");
+    return -1;
   }
-  fclose(proc_stat);
 
-  if (!value_valid)
+  fields_num = strsplit(ctxt, fields, STATIC_ARRAY_SIZE(fields));
+  if (fields_num != 2)
```

Comment:
Seems odd to hard code value on other line.  Maybe do it like this (untested):
```suggestion
  const int expected = STATIC_ARRAY_SIZE(fields);
  fields_num = strsplit(ctxt, fields, expected);
  if (fields_num != expected)
```